service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/CloneCreateCloudContextFlight.java

```diff
@@ -0,0 +1,107 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.workspace;
+
+import bio.terra.stairway.Flight;
+import bio.terra.stairway.FlightMap;
+import bio.terra.workspace.common.utils.FlightBeanBag;
+import bio.terra.workspace.common.utils.FlightUtils;
+import bio.terra.workspace.common.utils.RetryRules;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.job.JobMapKeys;
+import bio.terra.workspace.service.spendprofile.SpendProfile;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import bio.terra.workspace.service.workspace.model.CloudContext;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import bio.terra.workspace.service.workspace.model.GcpCloudContext;
+import java.util.Optional;
+import java.util.UUID;
+
+/** Flight for creating a cloud context for a cloned workspace */
+public class CloneCreateCloudContextFlight extends Flight {
+  public CloneCreateCloudContextFlight(FlightMap inputParameters, Object applicationContext) {
+    super(inputParameters, applicationContext);
+
+    // Flight Plan
+    // * Launch a flight to create a cloud context if necessary
+    // * Await the context flight
+    var flightBeanBag = FlightBeanBag.getFromObject(applicationContext);
+    var cloudRetryRule = RetryRules.cloud();
+    var longCloudRetryRule = RetryRules.cloudLongRunning();
+    var userRequest =
+        FlightUtils.getRequired(
+            inputParameters,
+            JobMapKeys.AUTH_USER_INFO.getKeyName(),
+            AuthenticatedUserRequest.class);
+    var sourceWorkspaceId =
+        FlightUtils.getRequired(
+            inputParameters, ControlledResourceKeys.SOURCE_WORKSPACE_ID, UUID.class);
+    var spendProfile =
+        inputParameters.get(WorkspaceFlightMapKeys.SPEND_PROFILE, SpendProfile.class);
+
+    addStep(new CreateCloudContextIdsForFutureStepsStep());
+
+    // Only create a cloud context if the source workspace has a cloud context
+    Optional<GcpCloudContext> gcpCloudContextOptional =
+        flightBeanBag.getGcpCloudContextService().getGcpCloudContext(sourceWorkspaceId);
+    if (gcpCloudContextOptional.isPresent()) {
+      SpendProfile gcpSpendProfile =
+          getSpendProfile(spendProfile, gcpCloudContextOptional.get(), flightBeanBag, userRequest);
+
+      addStep(
+          new LaunchCreateCloudContextFlightStep(
+              flightBeanBag.getWorkspaceService(),
+              CloudPlatform.GCP,
+              gcpSpendProfile,
+              ControlledResourceKeys.CREATE_GCP_CLOUD_CONTEXT_FLIGHT_ID),
+          cloudRetryRule);
+      addStep(
+          new AwaitCreateCloudContextFlightStep(
+              ControlledResourceKeys.CREATE_GCP_CLOUD_CONTEXT_FLIGHT_ID),
+          longCloudRetryRule);
```

Comment:
Curious why this is the long cloud retry rule vs the "regular" one for Azure on L80.